packages/components/src/box-control/vertical-horizontal-input-controls.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import UnitControl from './unit-control';
+import { LABELS } from './utils';
+import { Layout } from './styles/box-control-styles';
+
+const groupedSides = [ 'vertical', 'horizontal' ];
+
+export default function VerticalHorizontalInputControls( {
+	onChange = noop,
+	onFocus = noop,
+	onHoverOn = noop,
+	onHoverOff = noop,
+	values,
+	sides,
+	...props
+} ) {
+	const createHandleOnFocus = ( side ) => ( event ) => {
+		onFocus( event, { side } );
+	};
+
+	const createHandleOnHoverOn = ( side ) => () => {
+		if ( side === 'vertical' ) {
+			onHoverOn( {
+				top: true,
+				bottom: true,
+			} );
+		} else {
+			onHoverOn( {
+				left: true,
+				right: true,
+			} );
+		}
+	};
+
+	const createHandleOnHoverOff = ( side ) => () => {
+		if ( side === 'vertical' ) {
+			onHoverOff( {
+				top: false,
+				bottom: false,
+			} );
+		} else {
+			onHoverOff( {
+				left: false,
+				right: false,
+			} );
+		}
+	};
+
+	const createHandleOnChange = ( side ) => ( next ) => {
+		const nextValues = { ...values };
+
+		if ( side === 'vertical' ) {
+			nextValues.top = next;
+			nextValues.bottom = next;
+		}
+		if ( side === 'horizontal' ) {
```

Comment:
Bit of a nit, but in the `createHandleOnHoverOff` function the check for `horizontal` is implicit with the `else` statement, while in this function both `vertical` and `horizontal` are checked separately and explicitly.

Should we use the same approach (explicit or implicit) everywhere?

Reply:
Good point! I've cleaned this up to use the `explicit` approach, as it seems a bit clearer to me.

Reply:
Looks great, thank you!